### ISSUE TITLE
Squiz.Functions.FunctionDeclarationArgumentSpacing bug when no space between type hint and argument

### ISSUE THREAD

**Issue by dereuromark** [squizlabs/PHP_CodeSniffer#900]
This does only recognize the 2nd and 3rd:
```
 31 | ERROR   | [x] Expected 1 space between comma and argument "$c"; 0
    |         |     found
    |         |     (Squiz.Functions.FunctionDeclarationArgumentSpacing.NoSpaceBeforeArg)
 31 | ERROR   | [x] Expected 1 space between comma and argument "$o"; 0
    |         |     found
    |         |     (Squiz.Functions.FunctionDeclarationArgumentSpacing.NoSpaceBeforeArg)
```
in

```php
public function MissingParamTypeInDocBlock(array$a = null, callable$c, \ArrayObject$o, $foo = []) 
{
}
```

The `array$a` should be detected and fixed to `array $a`.
I assume it has to do with the beginning of the argument list.

The text is also a bit misleading, should maybe say "between typehint and argument". Or let the other rule take care of it, that does correctly use this terminology.

**Comment by dereuromark** [comment]
Actually, the reports it, but fixes it incorrectly:
```
 31 | ERROR   | [x] Expected 1 space between type hint and argument
    |         |     "$int"; 0 found
    |         |     (Squiz.Functions.FunctionDeclarationArgumentSpacing.SpacingAfterHint)
```
Result: `array  = null`, the variable is now missing.

**Comment by gsherwood** [comment]
There were a couple of bugs in there, and PSR2 shouldn't have been reporting any errors with this sniff.

With the fix, this code now produces the following errors for the Squiz standard:
```
FILE: temp.php
---------------------------------------------------------------------------------------------------------------------
FOUND 7 ERRORS AFFECTING 1 LINE
---------------------------------------------------------------------------------------------------------------------
 2 | ERROR | [x] Expected 1 space between type hint and argument "$a"; 0 found
 2 | ERROR | [x] Incorrect spacing between argument "$a" and equals sign; expected 0 but found 1
 2 | ERROR | [x] Incorrect spacing between default value and equals sign for argument "$a"; expected 0 but found 1
 2 | ERROR | [x] Expected 1 space between type hint and argument "$c"; 0 found
 2 | ERROR | [x] Expected 1 space between type hint and argument "$o"; 0 found
 2 | ERROR | [x] Incorrect spacing between argument "$foo" and equals sign; expected 0 but found 1
 2 | ERROR | [x] Incorrect spacing between default value and equals sign for argument "$foo"; expected 0 but found 1
---------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 7 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------------------------------------------
```
and the diff for the fixes is this:
```
--- temp.php
+++ PHP_CodeSniffer
@@ -1,4 +1,4 @@
 <?php
-public function MissingParamTypeInDocBlock(array$a = null, callable$c, \ArrayObject$o, $foo = [])
+public function MissingParamTypeInDocBlock(array $a=null, callable $c, \ArrayObject $o, $foo=[])
 {
 }
```
Thanks for reporting this.

**Comment by dereuromark** [comment]
:+1:
